### PR TITLE
Fix various sanity errors in plugins

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes some of the errors found in #71734 in ansible.builtin plugins.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/plugins/callback/tree.py
lib/ansible/plugins/connection/paramiko_ssh.py
lib/ansible/plugins/inventory/auto.py
lib/ansible/plugins/inventory/toml.py
lib/ansible/plugins/lookup/config.py
lib/ansible/plugins/lookup/csvfile.py
lib/ansible/plugins/lookup/file.py
lib/ansible/plugins/lookup/fileglob.py
lib/ansible/plugins/lookup/first_found.py
lib/ansible/plugins/lookup/indexed_items.py
lib/ansible/plugins/lookup/ini.py
lib/ansible/plugins/lookup/lines.py
lib/ansible/plugins/lookup/list.py
lib/ansible/plugins/lookup/password.py
lib/ansible/plugins/lookup/pipe.py
lib/ansible/plugins/lookup/random_choice.py
lib/ansible/plugins/lookup/sequence.py
lib/ansible/plugins/lookup/template.py
lib/ansible/plugins/lookup/together.py
lib/ansible/plugins/lookup/unvault.py
lib/ansible/plugins/lookup/url.py
lib/ansible/plugins/lookup/vars.py

### CODE REVIEW

File: lib/ansible/plugins/connection/paramiko_ssh.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 DOCUMENTATION = """
     author: Ansible Core Team
-    connection: paramiko
+    connection: paramiko_ssh
```
Comment (bcoca) [inline lib/ansible/plugins/connection/paramiko_ssh.py]: this is incorrect, it uses a alias in plugin_loader to show as paramiko, so the name should be kept

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/connection/paramiko_ssh.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/plugins/connection/paramiko_ssh.py]: Sure. (I guess this needs a hard-coded exception in the validator then.)

**Comment by felixfontein** [comment]
If I didn't overlook something, this fixes everything reported by #71734 except the `name` elements in options (see #71792):
```
lib/ansible/plugins/callback/default.py:0:0: invalid-documentation: DOCUMENTATION.options.check_mode_markers.name: extra keys not allowed @ data['options']['check_mode_markers']['name']. Got 'Show markers when running in check mode'
lib/ansible/plugins/callback/default.py:0:0: invalid-documentation: DOCUMENTATION.options.display_failed_stderr.name: extra keys not allowed @ data['options']['display_failed_stderr']['name']. Got 'Use STDERR for failed and unreachable tasks'
lib/ansible/plugins/callback/default.py:0:0: invalid-documentation: DOCUMENTATION.options.display_ok_hosts.name: extra keys not allowed @ data['options']['display_ok_hosts']['name']. Got "Show 'ok' hosts"
lib/ansible/plugins/callback/default.py:0:0: invalid-documentation: DOCUMENTATION.options.display_skipped_hosts.name: extra keys not allowed @ data['options']['display_skipped_hosts']['name']. Got 'Show skipped hosts'
lib/ansible/plugins/callback/default.py:0:0: invalid-documentation: DOCUMENTATION.options.show_custom_stats.name: extra keys not allowed @ data['options']['show_custom_stats']['name']. Got 'Show custom stats'
lib/ansible/plugins/callback/default.py:0:0: invalid-documentation: DOCUMENTATION.options.show_per_host_start.name: extra keys not allowed @ data['options']['show_per_host_start']['name']. Got 'Show per host task start'
lib/ansible/plugins/callback/junit.py:0:0: invalid-documentation: DOCUMENTATION.options.fail_on_change.name: extra keys not allowed @ data['options']['fail_on_change']['name']. Got 'JUnit fail on change'
lib/ansible/plugins/callback/junit.py:0:0: invalid-documentation: DOCUMENTATION.options.fail_on_ignore.name: extra keys not allowed @ data['options']['fail_on_ignore']['name']. Got 'JUnit fail on ignore'
lib/ansible/plugins/callback/junit.py:0:0: invalid-documentation: DOCUMENTATION.options.hide_task_arguments.name: extra keys not allowed @ data['options']['hide_task_arguments']['name']. Got 'Hide the arguments for a task'
lib/ansible/plugins/callback/junit.py:0:0: invalid-documentation: DOCUMENTATION.options.include_setup_tasks_in_report.name: extra keys not allowed @ data['options']['include_setup_tasks_in_report']['name']. Got 'JUnit include setup tasks in report'
lib/ansible/plugins/callback/junit.py:0:0: invalid-documentation: DOCUMENTATION.options.output_dir.name: extra keys not allowed @ data['options']['output_dir']['name']. Got 'JUnit output dir'
lib/ansible/plugins/callback/junit.py:0:0: invalid-documentation: DOCUMENTATION.options.task_class.name: extra keys not allowed @ data['options']['task_class']['name']. Got 'JUnit Task class'
lib/ansible/plugins/callback/junit.py:0:0: invalid-documentation: DOCUMENTATION.options.task_relative_path.name: extra keys not allowed @ data['options']['task_relative_path']['name']. Got 'JUnit Task relative path'
lib/ansible/plugins/callback/junit.py:0:0: invalid-documentation: DOCUMENTATION.options.test_case_prefix.name: extra keys not allowed @ data['options']['test_case_prefix']['name']. Got 'Prefix to find actual test cases'
lib/ansible/plugins/shell/sh.py:0:0: invalid-documentation: DOCUMENTATION.options.common_remote_group.name: extra keys not allowed @ data['options']['common_remote_group']['name']. Got 'Enables changing the group ownership of temporary files and directories'
```

**Review by bcoca — COMMENTED** [review]
changes look ok, just need to verify we don't have any code relying on the old structures.

**Comment by felixfontein** [comment]
I'm only aware of ansible-doc, and that part is already fixed (#71966).

**Comment by felixfontein** [comment]
@bcoca thanks a lot for reviewing and merging!

**Pull request merged**